src/box/vy_tx.c

```diff
@@ -591,12 +612,6 @@ vy_tx_prepare(struct vy_tx *tx)
 {
 	struct vy_tx_manager *xm = tx->xm;
 
-	if (tx->state == VINYL_TX_ABORT) {
```

Comment:
Turns out we can't remove this check. The problem is `vinyl_engine_prepare()` may yield before calling `vy_tx_prepare` (if there's not enough memory quota). If the transaction gets aborted in the meantime, we must not proceed with committing it. I will restore this check on the next push.

Reply:
Moved quota reservation code from the `engine_prepare` callback to the statement execution functions, see the new commit "vinyl: reserve memory quota on statement execution". The `engine_prepare` callback doesn't yield anymore.